### PR TITLE
Uplift third_party/tt-mlir to origin/main (3875027052558390aaaeee1dfb72b6f9c71aed49) 2026-05-15

### PR DESCRIPTION
**Pull request by vmilosevic**
This PR uplifts the third_party/tt-mlir submodule to origin/main

### CODE REVIEW

**Review by nvukobratTT — APPROVED** [review]

**Comment by vmilosevic** [comment]
### List of tenstorrent/tt-mlir commits since previous uplift:
[2ec8d1adb](https://github.com/tenstorrent/tt-mlir/commit/2ec8d1adb) d2m: don't always build AsmState for Allocator debugging (#8406)
[18901a2a9](https://github.com/tenstorrent/tt-mlir/commit/18901a2a9) Add canonicalization after WA and LayoutDecomp (#8383)
[16d4bb34a](https://github.com/tenstorrent/tt-mlir/commit/16d4bb34a) [d2m] pipe clean dram to dram flow and add elementwise op tests (#8398)
[0e2755180](https://github.com/tenstorrent/tt-mlir/commit/0e2755180) d2m: beam search on eltwise op reblocking to restrict search space (#8409)
[d20301bd0](https://github.com/tenstorrent/tt-mlir/commit/d20301bd0) [D2M] Add vgm core_index annotation pass later in pipeline, remove logic from lowering passes (#8405)
[53dc2bb1b](https://github.com/tenstorrent/tt-mlir/commit/53dc2bb1b) [builder] Add @tag/@parse/@split for ttir.fill_cache, ttir.gelu, ttir.mesh_partition (#8305)
[28c2bbaa1](https://github.com/tenstorrent/tt-mlir/commit/28c2bbaa1) [D2M] Move `InsertDstRegisterAccess*` transform into a dedicated subdir (#8420)
[2e2a8f074](https://github.com/tenstorrent/tt-mlir/commit/2e2a8f074) Support for ttnn-mode in builder (#8397)
[d3f885724](https://github.com/tenstorrent/tt-mlir/commit/d3f885724) [d2m] Infer unified-thread synchronized region from remote ops (#8394)
[84836895e](https://github.com/tenstorrent/tt-mlir/commit/84836895e) [D2M] enable l1 acc by default (#8387)
[eb9005fa3](https://github.com/tenstorrent/tt-mlir/commit/eb9005fa3) [TTIR] Narrow RankNormalization scoping to skip only const_eval helpers (#8414)
[b1883766e](https://github.com/tenstorrent/tt-mlir/commit/b1883766e) [D2M] Remove OOB Value from tensor types (#8357)
[9f01a59c2](https://github.com/tenstorrent/tt-mlir/commit/9f01a59c2) [d2m] move reduction-fusion test to the correct folder (#8404)
[dd0431898](https://github.com/tenstorrent/tt-mlir/commit/dd0431898) [CI] Autosplit support for builder jobs (#8392)
[7fce00fbe](https://github.com/tenstorrent/tt-mlir/commit/7fce00fbe) [L1SpillManagement] Drop outputLayout hint for multi-output ops in extractOpConfigFromIR (#8337)

**Review by nvukobratTT — APPROVED** [review]

**Comment by vmilosevic** [comment]
### List of tenstorrent/tt-mlir commits since previous uplift:
[387502705](https://github.com/tenstorrent/tt-mlir/commit/387502705) [Logger] Add FusionValidator logger type (#8430)
[cad4698a5](https://github.com/tenstorrent/tt-mlir/commit/cad4698a5) Fix TopK workaround to use tile padding instead of power-of-2 (#8434)
[7e9bbc6a2](https://github.com/tenstorrent/tt-mlir/commit/7e9bbc6a2) Initial Quasar system descriptor support (#8396)
[f8e1738dd](https://github.com/tenstorrent/tt-mlir/commit/f8e1738dd) [StableHLOToTTIR] Handle 3D attention_mask in SDPA composite lowering (#8416)
[b1edfd8d7](https://github.com/tenstorrent/tt-mlir/commit/b1edfd8d7) [golden] add chisel mappings for TTNN op goldens (#8354)
[c2897923d](https://github.com/tenstorrent/tt-mlir/commit/c2897923d) Move TopK fusing to TTIR and rename FusionValidation to OpValidation (#8299)
[2ec8d1adb](https://github.com/tenstorrent/tt-mlir/commit/2ec8d1adb) d2m: don't always build AsmState for Allocator debugging (#8406)
[18901a2a9](https://github.com/tenstorrent/tt-mlir/commit/18901a2a9) Add canonicalization after WA and LayoutDecomp (#8383)
[16d4bb34a](https://github.com/tenstorrent/tt-mlir/commit/16d4bb34a) [d2m] pipe clean dram to dram flow and add elementwise op tests (#8398)
[0e2755180](https://github.com/tenstorrent/tt-mlir/commit/0e2755180) d2m: beam search on eltwise op reblocking to restrict search space (#8409)
[d20301bd0](https://github.com/tenstorrent/tt-mlir/commit/d20301bd0) [D2M] Add vgm core_index annotation pass later in pipeline, remove logic from lowering passes (#8405)
[53dc2bb1b](https://github.com/tenstorrent/tt-mlir/commit/53dc2bb1b) [builder] Add @tag/@parse/@split for ttir.fill_cache, ttir.gelu, ttir.mesh_partition (#8305)
[28c2bbaa1](https://github.com/tenstorrent/tt-mlir/commit/28c2bbaa1) [D2M] Move `InsertDstRegisterAccess*` transform into a dedicated subdir (#8420)
[2e2a8f074](https://github.com/tenstorrent/tt-mlir/commit/2e2a8f074) Support for ttnn-mode in builder (#8397)
[d3f885724](https://github.com/tenstorrent/tt-mlir/commit/d3f885724) [d2m] Infer unified-thread synchronized region from remote ops (#8394)
[84836895e](https://github.com/tenstorrent/tt-mlir/commit/84836895e) [D2M] enable l1 acc by default (#8387)
[eb9005fa3](https://github.com/tenstorrent/tt-mlir/commit/eb9005fa3) [TTIR] Narrow RankNormalization scoping to skip only const_eval helpers (#8414)
[b1883766e](https://github.com/tenstorrent/tt-mlir/commit/b1883766e) [D2M] Remove OOB Value from tensor types (#8357)
[9f01a59c2](https://github.com/tenstorrent/tt-mlir/commit/9f01a59c2) [d2m] move reduction-fusion test to the correct folder (#8404)
[dd0431898](https://github.com/tenstorrent/tt-mlir/commit/dd0431898) [CI] Autosplit support for builder jobs (#8392)
[7fce00fbe](https://github.com/tenstorrent/tt-mlir/commit/7fce00fbe) [L1SpillManagement] Drop outputLayout hint for multi-output ops in extractOpConfigFromIR (#8337)
